app/scripts/models/account.js

```diff
@@ -55,6 +55,7 @@ define(function (require, exports, module) {
     // move forward, very hard to move back.
     sessionTokenContext: undefined,
     uid: undefined,
+    verificationMethod: undefined,
```

Comment:
It seems like it! Here's the thing: on the server side, we aren't going to really have saved exactly if a message was sent by email or by push. The token will have been generated either way, and the `sessionToken` will know it's not verified, but the exact method the link was sent to a user is only reported in the response from `/account/login`, by the `verificationMethod`.

The content-server sees that the first time, and could remember it (currently it does not). Since it doesn't, if the content-server loads again and tries to submit without a password (it had a persisted `sessionToken`), it just assumes the method was email. If it persists the `verificationMethod` also, then it will be able to know which page it should load up.